src/arithmetic/agg_funcs.c

```diff
@@ -581,7 +581,7 @@ int __agg_collectStep(AggCtx *ctx, SIValue *argv, int argc) {
 
 	SIValue value = argv[0];
 	if(value.type == T_NULL) return AGG_OK;
-	SIArray_Append(&ac->list, value);
+	SIArray_Append(&ac->list, SI_ShareValue(value));
```

Comment:
Can you move this call to the invoking function? Usually, we share the value from the value owner and not on the value receiver, or use `SIValue_MakeVolatile`
since this is a very delicate resource handling, please add a comment for the reason and meaning of making this value volatil